src/sardana/macroserver/msmacromanager.py

```diff
@@ -1290,9 +1319,56 @@ def __runXMLMacro(self, xml):
     _runXMLMacro = __runXMLMacro
 
     def runMacro(self, macro_obj):
+
         name = macro_obj._getName()
         desc = macro_obj._getDescription()
         door = self.door
+        macro_obj = macro_obj
+
+        if macro_obj.getParentMacro() is None:
+            try:
+                logging_onoff = macro_obj.getEnv("LogMacro")
+            except:
+                logging_onoff = False
+            if logging_onoff:
+                try:
+                    self.logging_mode = macro_obj.getEnv("LogMacroMode")
+                except:
+                    self.logging_mode = 0
+                try:
+                    logging_path = macro_obj.getEnv("LogMacroDir")
+                except:
+                    macro_obj.setEnv("LogMacroDir", "/tmp")
```

Comment:
Also two comments:
- Do you think it is necessary to set the environment variable, or just silently use the default one?
- I'm not sure, but this may not work on Windows. What if you use `os.path.join` and `os.sep`. For an example look in `sardana.tango.core.util.prepare_logging`.

Reply:
I have changed the building of the path according to your suggestion. But I still kept the set of the
MacroLogDir environment. I am not sure if it is better to set it or not ... But perhaps it is useful for the users ... if they activate the logging and they don't know which the default directory is, they will see it listing the environment. What do you think?

Reply:
An alternative would be to print the info message after enabling the logging where the logs will be saved. If you think it is not enough then leave the environment setting.

Reply:
Hi @reszelaz, I implemented now like this ... but it is a little bit annoying because one gets a message like:
Log file saved in /tmp

everytime a macro is started. One could put it in debug instead of info ... If it is enough ... the users would say.

Reply:
I agree it would be too much to print it everytime. So let's leave the environment variable setting.